### DISCUSSION TITLE
Help visualize or verify schema design with `export to` or `verify with` functionality on DBML

### DISCUSSION THREAD

**Discussion by curiouslaw** [simolus3/drift#2829] [Ideas]
Having a quick way to visualize the entire schema with just a simple copy-paste a text file (e.g., directly from Github repo web) would be a great help for developer and data people. Thus suggesting to achieve this with the help of [DBML syntax](https://github.com/holistics/dbml). A function for exporting the current schema into the DBML syntax would be helpful.

As an extension, the feature to verify if the current schema version match the visualized schema design draft (with the DBML syntax) would be great.

**Comment by simolus3** [comment]
> A function for exporting the current schema into the DBML syntax would be helpful.

You can do this with [runtime schema inspection](https://drift.simonbinder.eu/docs/dart-api/schema_inspection/), which gives you a full description of the database schema at runtime. That should be enough to create a DBML syntax of the schema as drift would expect it.

> the feature to verify if the current schema version match the visualized schema design draft (with the DBML syntax) would be great

Drift already has [some tools](https://drift.simonbinder.eu/docs/migrations/#verifying-a-database-schema-at-runtime) to validate the integrity of the database schema. If you can turn `CREATE TABLE` statements into DBML, you could also select from `sqlite_schema` and compare the DBML you get from there with what you expect.

**Reply by curiouslaw** [reply]
Well noted and thanks for the information!
